File: data_structures/stacks/Java/Stack.java
public class Stack<T> {

	int size;
	int top = -1;
	T[] stack;

	// generic Stack inisialisation
	@SuppressWarnings("unchecked") // ignores compiler warnings
	public Stack(int size) {
		this.size = size;
		this.stack = (T[]) new Object[size];
	}

	// addin an element at the top of the stack
	public void push(T num) {
		stack[++top] = num;
		System.out.println("Pushed " + num);
	}

	// removing the top element
	public void pop() {
		System.out.println("Poped " + stack[top--]);
	}

	// addressing the top element
	public T top() {
		System.out.println("Top element is " + stack[top]);
		return stack[top];
	}

	// checking if stack is empty
	public boolean isEmpty() {
		if (top == -1) {
			System.out.println("Stack is empty");
			return true;
		}
		System.out.println("Stack is not empty");
		return false;
	}

	// checking if stack is full
	public boolean isFull() {
		if (top == size - 1) {
			System.out.println("Stack is full");
			return true;
		}
		System.out.println("Stack is not full");
		return false;
	}
}

File: data_structures/stacks/C/stack.c
#include<stdio.h>

int size=10;
int stack[10];
int top = -1;
int empty()
{
  if(top==-1)
  return 1;
  else
  return 0;
}
int full()
{
  if(top==10)
  return 1;
  else
  return 0;
}
void push(int data)
{
  top++;
  stack[top]=data;
}
void pop()
{
  top--;
}
void display()
{
  for(int i=0;i<=top;i++)
  printf("%d\n",stack[i]);
}
void main()
{
  int ch,val;
  do
  {
  printf("1.push\n");
  printf("2.pop\n");
  printf("3.display\n");
  printf("4.exit\n");
  printf("enter choice");
  scanf("%d",&ch);
  switch (ch) {
    case 1:
    if(full()==1)
    printf("OOPS!..stack is full already\n");
    else
    {
      printf("enter data");
      scanf("%d",&val);
      push(val);
    }
    break;
    case 2:
    if(empty()==1)
    printf("OOPS!..stack is empty\n");
    else
    pop();
    break;
    case 3:
    display();
    break;
    case 4:
    break;
    default :
    printf("enter correct option");
  }}
  while(ch!=4);
}
